core/windows_and_tabs/window_snap.py

```diff
@@ -310,6 +338,30 @@ def snap_app(app_name: str, position: RelativeScreenPos):
         _bring_forward(window)
         _snap_window_helper(window, position)
 
+    def snap_split_two(app_one: str, app_two: str):
+        """Split the screen between two applications."""
+        window = _get_app_window(app_one)
+        window_two = _get_app_window(app_two)
+        _snap_window_helper(window_two, _snap_positions["right"])
+        _snap_window_helper(window, _snap_positions["left"])
+        window_two.focus()
+        window.focus()
+
+    def snap_split_three(
+        positions: list[RelativeScreenPos], app_one: str, app_two: str, app_three: str
+    ):
```

Comment:
You can just make this one single action with lots of application parameters, and just make the ones greater than 2 optional:

```suggestion
    def snap_layout(
        positions: list[RelativeScreenPos],
        app_one: str,
        app_two: str,
        app_three: Optional[str],
        app_four: Optional[str],
        app_five: Optional[str]
    ):
```
Might as well support up to 5 in case people want to make more interesting layouts.

I don't think there's an explicit `null` in Talonscript but the shorter-n versions can just leave those parameters off:

```
user.snap_layout("abc", "def", "jkl")
```

`app_four` and `app_five` will thus be `None`.

That way you don't have to duplicate the action for each number of applications.

Also I think it'd be helpful to give this functionality a useful name, to distinguish it from the existing single-window snapping functionality. Since "split" is itself a layout I think it should probably have a broader name like "layout" or "arrangement" (not strongly opinionated on the name itself, just that it has one). Do any of those sound good to you, or perhaps do you have a better one? 😁